### ISSUE TITLE
Fragment links are not checked correctly when pointing to folder having index.html

### ISSUE THREAD

**Issue by svlapin** [Munter/hyperlink#150]
While trying to understand why links on webpack.js.org got broken without linter having caught them (e.g. https://github.com/webpack/webpack.js.org/pull/2929), I found out that fragment links are not checked correctly then they point to a folder having `index.html` inside.

To illustrate the issue, I created `testdata/fragmentIdentifier/test/index.html`
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Document</title>
</head>
<body>
</body>
</html>
```

and appended
```diff
diff --git a/testdata/fragmentIdentifier/index.html b/testdata/fragmentIdentifier/index.html
index 5ec7aff..2e12ebd 100644
--- a/testdata/fragmentIdentifier/index.html
+++ b/testdata/fragmentIdentifier/index.html
@@ -26,5 +26,7 @@
     <a href="page.html#broken-two"></a>
     <a href="page.html#broken-three"></a>
 
+    <a href="/test#definitely-broken"></a>
+
 </body>
 </html>
```

Running
```sh
./lib/cli.js testdata/fragmentIdentifier/index.html
```
doesn't report `test#definitely-broken` failed. It only does if `index.html` is used in the link, like

```diff
diff --git a/testdata/fragmentIdentifier/index.html b/testdata/fragmentIdentifier/index.html
index 5ec7aff..fb497aa 100644
--- a/testdata/fragmentIdentifier/index.html
+++ b/testdata/fragmentIdentifier/index.html
@@ -26,5 +26,7 @@
     <a href="page.html#broken-two"></a>
     <a href="page.html#broken-three"></a>
 
+    <a href="/test/index.html#definitely-broken"></a>
+
 </body>
 </html>
```

I'd be happy to come up with a PR, but wasn't able to identify the root cause. Ended up finding that condition [relation.to.type === 'Html'](https://github.com/Munter/hyperlink/blob/master/lib/index.js#L519) is not fulfilled in this case and so `relation.to.incomingFragments` are not populated.

**Comment by papandreou** [comment]
Thanks for the report and the reproduction case! I've opened a PR with a fix: #151

**Comment by papandreou** [comment]
Fixed in 4.1.1
